ndnsd/discovery/service-discovery.cpp

```diff
@@ -321,36 +306,60 @@ ServiceDiscovery::processSyncUpdate(const std::vector<ndnsd::SyncDataInfo>& upda
 
 template<ndn::encoding::Tag TAG>
 size_t
-ServiceDiscovery::wireEncode(ndn::EncodingImpl<TAG>& encoder,
-                             const std::string& info, int status) const
+ServiceDiscovery::wireEncode(ndn::EncodingImpl<TAG>& encoder, const std::string& info) const
 {
   size_t totalLength = 0;
   totalLength += prependStringBlock(encoder, tlv::ServiceInfo, info);
-  totalLength += prependNonNegativeIntegerBlock(encoder, tlv::ServiceStatus, status);
+  totalLength += prependNonNegativeIntegerBlock(encoder, tlv::ServiceStatus, m_serviceStatus);
   totalLength += encoder.prependVarNumber(totalLength);
   totalLength += encoder.prependVarNumber(tlv::DiscoveryData);
 
   return totalLength;
 }
 
 const ndn::Block&
-ServiceDiscovery::wireEncode(const std::string& info, int status)
+ServiceDiscovery::wireEncode()
 {
   if (m_wire.hasWire())
```

Comment:
If hasWire, you should simply resend. The below string should be saved in a class member and then reset wire when this string is modified/updated.